src/main/kotlin/com/exactpro/th2/common/microservice/ApplicationContext.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2023 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.exactpro.th2.common.microservice
+
+import com.exactpro.th2.common.schema.factory.AbstractCommonFactory
+
+/**
+ * @param registerResource: register all resources which must be closed in reverse registration order.
+ * @param onPanic: call this method when application can not correct work anymore.
+ */
+class ApplicationContext(
+    val commonFactory: AbstractCommonFactory,
+    val registerResource: (AutoCloseable) -> Unit,
```

Comment:
I think it would be convenient to provide a name for the resource. It will make logging messages more informative. I also think that we should use `BiConsumer<T, U>` here (it will be more convenient for Java)
```suggestion
    val registerResource: BiConsumer<String, AutoCloseable>,
```